src/hu_crosshair.c

```diff
@@ -0,0 +1,178 @@
+//
+//  Copyright (C) 1999 by
+//  id Software, Chi Hoang, Lee Killough, Jim Flynn, Rand Phares, Ty Halderman
+//  Copyright (C) 2023-2024 Fabian Greffrath
+//
+//  This program is free software; you can redistribute it and/or
+//  modify it under the terms of the GNU General Public License
+//  as published by the Free Software Foundation; either version 2
+//  of the License, or (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+// DESCRIPTION:  Heads-up displays
+//
+//-----------------------------------------------------------------------------
+
+#include "doomstat.h"
+#include "d_items.h"
+#include "hu_crosshair.h"
+#include "hu_stuff.h"
+#include "m_swap.h"
+#include "p_map.h"
+#include "p_mobj.h"
+#include "r_data.h"
+#include "r_main.h"
+#include "r_state.h"
+#include "st_stuff.h"
+#include "v_fmt.h"
+#include "v_video.h"
+
+static player_t *plr;
+
+int hud_crosshair;
+
+// Crosshair
+
+crosstarget_t hud_crosshair_target;
+boolean hud_crosshair_health;
+boolean hud_crosshair_lockon; // [Alaux] Crosshair locks on target
+int hud_crosshair_color;
+int hud_crosshair_target_color;
+
+typedef struct
+{
+  patch_t *patch;
+  int w, h, x, y;
+  byte *cr;
+} crosshair_t;
+
+static crosshair_t crosshair;
+
+const char *crosshair_lumps[HU_CROSSHAIRS] =
+{
+  NULL,
+  "CROSS00", "CROSS01", "CROSS02", "CROSS03",
+  "CROSS04", "CROSS05", "CROSS06", "CROSS07",
+  "CROSS08"
+};
+
+const char *crosshair_strings[HU_CROSSHAIRS] =
```

Comment:
At some point before this PR, these names stopped showing up:

![Untitled](https://github.com/fabiangreffrath/woof/assets/56656010/ab37df98-cd78-4c25-beb8-00c120d36a36)

Reply:
Formerly there was a check whether the crosshair patch lumps were among the predefined lumps. The code that runs more would only execute if they were not.

Reply:
Okay I see: https://github.com/fabiangreffrath/woof/pull/1765/files#diff-9d286a8abc73359c66378a9db33bc11b89cd2b60440c1f43a560328914e293d7R77

It reads from woof.pk3 so it's always true I'm gussing.

Reply:
We should check if it's from a PWAD instead. Woof.pk3 is not considered a PWAD.

Reply:
The point is that the lump names are trivial and a PWAD could provide a lump of that name that's not a patch, and so the crosshair code would always crash.

Reply:
We can add new PWAD markers (e.g. `CH_START` `CH_STOP`) or create a new folder for it (`base/all-all/crosshairs`). ZDoom variant: [wiki](https://zdoom.org/wiki/Crosshair#Creating_your_own_crosshairs) (not compatible with our format).